app/aboutdialog.cpp

```diff
@@ -71,14 +71,15 @@ AboutDialog::AboutDialog(QWidget *parent)
     QStringList licenseStr {
         QStringLiteral("<h1 align='center'><b>%1</b></h1>").arg(tr("Your Rights")),
         QStringLiteral("<p>%1</p><p>%2</p><ul><li>%3</li><li>%4</li><li>%5</li><li>%6</li></ul>").arg(
-            tr("%1 is released under the MIT License."), // %1
+            tr("%1 is MIT licensed."), // %1
+
             tr("This license grants people a number of freedoms:"), // %2
             tr("You are free to use %1, for any purpose"), // %3
             tr("You are free to distribute %1"), // %4
             tr("You can study how %1 works and change it"), // %5
             tr("You can distribute changed versions of %1") // %6
         ).arg(QStringLiteral("<i>%1</i>")),
-        QStringLiteral("<p>%1</p>").arg(tr("The MIT license guarantees you this freedom. Nobody is ever permitted to take it away.")),

```

Comment:
This is very misleading. Only the license text needs to be retained. The MIT license is not copyleft.

Reply:
Yeah The MIT license is not copyleft, but it did guarantee users this freedom as long as the user gets the software and source code, and for the copy of software/source code, the freedom (guarantee by MIT license) is not permitted to take away by anybody else, so IMO this is okay to leave it here. Is that correct?

I'm not sure if I understand this correctly, let me know if it's wrong.

Reply:
> This is very misleading. ... The MIT license is not copyleft.

[The four essential freedoms](https://www.gnu.org/philosophy/free-sw.html.en) doesn't mean the software should have a copyleft license. The MIT license actually grants users these rights (and MIT is also an [FSF-approved license](https://www.gnu.org/licenses/license-list.en.html#Expat)), so I think this text is not misleading at all and I prefer to retain this text if I understand correctly.

Reply:
True, it is this part "Nobody is ever permitted to take it away." which is indistinguishable from copyleft. In some sense of the wording "it" means the freedoms, which can be "taken away" since it is permissible. The MIT license itself, if that is "it", can also effectively be taken away. I already think "guarantees" is strong, but I don't see how "Nobody is ever", which couldn't be stronger (and probably doesn't account for it entering the public domain?) etc. adds anything.

Reply:
> In some sense of the wording "it" means the freedoms, which can be "taken away" since it is permissible.
> The MIT license itself, if that is "it", can also effectively be taken away.

That's not true, the MIT license allowed gives user those essential freedoms and there is no way to revoke the license which the user is already gotten (unless user abuse the license itself). If someone else takes a copy of this software and turn their forked version into non-free software, that won't affect the other users who get their Expat/MIT licensed copy, since the Expat/MIT license grants users the four essential freedoms, nobody is ever permitted to take it away if the user already gots their license which grants users these freedoms. (FSF won't approve the license if someone else could "take away" these freedoms)

I'm not sure why you think the freedoms can be taken away from users if the software is using a not copyleft license, If you think someday the copyright holder (I) can change the license and turn it into a non-free software, let's assume I did so, if the users can still find and read that text, it means that user already got an Expat/MIT licensed copy of this software, which means they still have the freedom to use the copy they got since the license grants the user these freedoms. So as you can see, even if I can do that will still not able take away the freedom from the users who already get a copy of the software (along with a valid Expat/MIT license), I think writing "Nobody is ever permitted to take it away" inside the software is okay.

Is that correct?

Reply:
"which the user is already gotten" isn't mentioned, and crucially is thing you have to already know about to make that work.
"Nobody is ever permitted to take it away" as it relates to "it", also applies for what _the source code_ is as it exists, not merely how you got it. The word "guarantee" alone is probably unwarranted, but certainly if it also covers this (without the disclaimer of distribution).

Similarly and furthermore,
"You can distribute changed versions of %1" doesn't hold true for
"Nobody is ever permitted to take it away."

I think the Freedoms are the ability to do these things. When those things are no longer possible, I don't think it matters that distribution is what granted the ability in the first place. Distribution as it relates to licensing is just a function of copyright. I don't care for copyright law, and unfortunately licenses have to be longer because the world of "abusing the license" isn't ideal. Having to deal in non-ideals, that is my take, and it is hard enough to convey.

"users who already get a copy of the software" I think _the software_ applies to what the software is, not what someone decided on as they distributed it to me. Yes, there needs to be a meaningful contribution to distribute it under a different license (I think), but that is semantics in terms of what user freedoms to the version of the software that matters.

As an example of the ambiguity, lets say you own the rights to some land. If I lease it for 10000 years, you don't own it less, but the difference between selling it to me and being able to do anything with the ownership vanishes in a way that doesn't warrant. "You keep owning your land just like before." It is true in some sense, but it isn't a good application of language unless you want to mislead.

Reply:
> What the MIT license doesn't do is even grant, much less guarantee the additional copyleft.

But what the text in the application referred to is **the four essential freedoms** which provided by the Expat/MIT license. It's not related to copyleft at all.

> isn't mentioned, and crucially is the thing you have to already know about to make that work.

I don't think we should mention it again since if the user can read that text, obviously, the user already got their Expat/MIT licensed copy. The text itself is inside the application, not in somewhere like a homepage/official website. The text is intended to tell the user this software is free software that comes with a free (as in speetch) license (Expat/MIT) which grantee user freedom they already gotten.

> there is some ambiguity over what "it" means, but in either case it is shifty. "Freedom" as a whole can cover many concepts, so it could very well be all the above freedoms.

We are focus on the four essential freedoms, which is perfectly clear MIT license grantee users these freedoms.

> "You can distribute changed versions of %1" doesn't hold true for
> "Nobody is ever permitted to take it away."

Nobody is ever permitted to take it (for the user, the right to redistribute their changed version) away, that's always true. Elsewhere the original MIT licensed software won't be free software.

> I think the Freedoms are the ability to do these things. When those things are no longer possible, I don't think it matters that distribution is what granted the ability in the first place.

Yeah, and that's what I've said, users always have their right to do that if they got their Expat/MIT licensed copy, they even have their right to turn it into non-free software if needed, I'm not able to take their freedom away. But to make it perfectly clear, these rights don't mean the license should be copylefted.

> *I think* the software applies to what the software is

Since the text itself is inside the application, it refers to the copy they got. What you've said will make a scene only if I put that text in a general public place to let other people who haven't got their license copy. And also, since it's free software, even if I made my decision to turn it closed-source or something else, other people who got their free (as in speech) copy still have their rights to maintain a free libre fork.

The text is explaining how the license working and what did the license grants to the user. It's all about license text in a tab dedicated to licensing, of course, it's all related to licensing itself than covering the whole project management.

> "You keep owning your land just like before."

**That's how (free/libre) licenses work**. and that's why the explanation text meant to be related to the copy they got instead of refers to the whole project.

> It is true in some sense, but it isn't a good application of language unless you want to mislead.

A copylefted license adding extra limitation about how users can use the source code, but it's still true the license itself can only apply to the copy they got. It's not misleading at all. We are restricted to explain the license terms here, not about how the project will be in the future.

> There is a near zero chance it won't get even worse in translation.

The actual valid text in legal is still the original license text(which provided below the explanation text. If there is an error in a translation, such an error needs to be corrected. If you think the explain text may be wrong during translation, a disclaimer text can also be added, but since the original text is valid, I prefer to keep it here instead of removing it.

I don't think that could be an issue since Krita also using that text to tell users they have these freedoms. I take the idea to my project here since it generally applies to all free software, no matter it's using a copylefted license or a lax permissive license.

---------------

By the way, I can also explain why I choose a lax permissive license for this project instead of a copylefted license. The project itself is pretty simple and stupid, it provides users a way to open images. The project is intended to be tiny and small and won't have a bunch of features like other image management software, as [GNU also doesn't recommend to use a copylefted license for the small project](https://www.gnu.org/licenses/gpl-faq.en.html#WhatIfWorkIsShort), I choose Expat/MIT license to grant user more freedom than make the project copylefted.

Another thing is, if I made it GPL/LGPL, once other contributors contribute copyrightable source code into the codebase, I'll no longer have the right to redistribute the software to somewhere like Apple Store or other similar platforms due to the license limitation unless a CLA is required to be signed by contributors. That's really not what I want, if I made it Expat/MIT licensed, not only I could do this, other people can also do this with no strings attached as long as they are not abusing the license term.

In a conclusion, I prefer to keep the text here to tell users their rights grants by the license which comes with their copy of the free software, instead of removing it.

Reply:
The rights are still spelled out. They are not guaranteed, nor are they irrevocable. The changed version tells the user the freedoms are in the license, nothing more.

What the license is and what the FSF thinks about it is beside the point.

Reply:
> They are not guaranteed, nor are they irrevocable

That's so untrue since it IS the license that guaranteed these freedoms and MIT is irrevocable (at least in the US, UK, EU, and mainland China AFAIK). Else it won't be able to be a free software license if the copyright holder can revoke someone else's license.

Please notice, relicense is a different thing, which doesn't affect the users who already got a valid license, software can be relicensed by the copyright holder doesn't mean the license which already grants to the user is not irrevocable.